app/controllers/manage_metadata_controller.rb

```diff
@@ -191,6 +191,56 @@ def set_service_information
     end
   end
 
+  def set_tool
```

Comment:
I think my biggest concern with metaprogramming to write these functions is figuring out where they are to make modifications.  If the debugging tool says the error is in 'set_tool', we should be able to search 'set_tool' and find a thread to pull on.  I feel the complexity of the V/S/T ecosystem is already high (already has metaprogramming + the schema code).  I would definitely like to hear other opinions on this though.

Reply:
I don't think we need to resolve the metaprogramming issue in this PR, but we should figure it out before the end of the published Tool records functionality. I understand your concern, and these methods are actually in a somewhat confusing place because they are used by both drafts and published records so ended up way up here in the chain.

I leaning slightly towards refactoring these methods with metaprogramming similar to some of the base drafts methods, and adding more comments explaining these methods and also in the v/s/t controllers to say what method they are using and where.

Reply:
I do feel that readability will take a hit by doing this, however, I think this should be weighed against the result of reduced lines of code (if that is in fact the result) which _should_ result in less code to support. Metaprogramming needs a good working example before it can be adopted across the board by other developers. I'm in favor of seeing an example, seeing documentation, and if the solution is radically different code, a wiki write up on it the practice. Having these things in place will increase the chances that the code is readable and increase the chances that we can take advantage of the technique.  As for this ticket, if you can get the change done in time for sprint end, the lets see it, otherwise a new ticket. Maybe we just leave a todo note for now?

Reply:
I don't think we'll forget about it, but am happy to add a todo. I don't think we need a new ticket, all the other Tools tickets to enable those action links should use these methods, plus at least one will need to tweak these methods. I suggest we tackle it in one of those upcoming tickets.